flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -1266,7 +1269,12 @@ public void goToRestarting(
                         forcedRestart,
                         userCodeClassLoader,
                         failureCollection));
-        numRestarts++;
+
```

Comment:
Prior to the change the numRestarts was all the restarts (including the rescales). After the fix, this metric now only reports restarts that were not rescales.  `numRestarts` implies all restarts to me. I suggest we always increment `numRestarts` as before, then add 

```
if (failureCollection.isEmpty()) {
            numRescales++;
}
```

We would need to amend the documentation accordingly to say numRestarts includes all restarts including rescales.

Reply:
Good point. We shouldn't change the current metric values. It's fine to add a new one though.

Reply:
@davidradl @mxm 
Thanks for the suggestion. It is reasonable to consider the metrics definition back-compatibility. Revised accordingly.